src/index.js

```diff
@@ -92,13 +92,20 @@ class CID {
     return this.buffer
   }
 
+  toV0String () {
+    if (this.version === 0) {
+      return mh.toB58String(this.multihash)
+    }
+
+    throw new Error(`Not supported on version: ${this.version}`)
+  }
+
   /* defaults to base58btc */
   toBaseEncodedString (base) {
     base = base || 'base58btc'
 
     switch (this.version) {
       case 0:
-        return mh.toB58String(this.multihash)
       case 1:
```

Comment:
Here you go :)

```
const mh = multihashing('sha-256', new Buffer('hello world')
const cid1 = new CID(mh)
const cid2 = new CID(1, 'dag-pb', mh)
const cid3 = new CID(0, 'dag-pb', mh)

console.log(cid1.toBaseEncodedString())
// Qm...  what we are used to see, a base encoded string of just the multihash because this CID is V0
console.log(cid2.toBaseEncodedString())
// a base encoded string of a CIDV1, meaning cid version, multicodec and multihash
console.log(cid3.toBaseEncodedString())
// exactly the same as cid1
```

This way, you can pass a CID to bitswap (for example) and bitswap will always do cid.toBaseEncodedString() to reference it around, without even having to think if it is V0 or V1 (as it is now).

The only current limitation is that CIDV0 was being forced to Base58, which is something that we _recently learned_ we don't need to force and it will actually be useful to support other base encodings for CIDV0.